pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -353,3 +480,83 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
+	pvcAnn, err := c.getPVCAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get pvc annotations")
+		return false, err
+	}
+
+	if val, found := pvcAnn[common.AnnIgnoreInaccessiblePV]; found && val == "yes" {
+		log.Infof("Found %s annotation on pvc for volume %s set to yes", common.AnnIgnoreInaccessiblePV, volumeID)
+		//Check if volume is inaccessible
+		log.Debugf("Checking volume health on CNS volume: %s", volumeID)
+		vol, err := common.QueryVolumeByID(ctx, volumeManager, volumeID)
+		if err != nil {
+			log.Errorf("failed to query CNS volume")
+			return false, err
+		}
+
+		if vol.HealthStatus != string(pbmtypes.PbmHealthStatusForEntityUnknown) {
+			volHealthStatus, err := convertVolumeHealthStatus(vol.HealthStatus)
+			if err != nil {
+				log.Errorf("invalid health status: %s for volume: %s", vol.HealthStatus, vol.VolumeId.Id)
+				return false, err
+			}
+			log.Infof("CNS volume health is: %s", volHealthStatus)
+
+			annotations := make(map[string]string)
+			// If CNS volume health doesn't match with volume health annotation on pvc,
+			// mark volume health on pvc to be updated.
+			if val, found := pvcAnn[common.AnnVolumeHealth]; !found || val != volHealthStatus {
+				annotations[common.AnnVolumeHealth] = volHealthStatus
+			}
+			// If volume is inaccessible, it can be fake attached.
+			// Mark fake-attached annotation on pvc to be updated.
+			if volHealthStatus == common.VolHealthStatusInaccessible {
+				annotations[common.AnnFakeAttached] = "yes"
+			}
+			// Update annotations
+			if err := c.updatePVCAnnotations(ctx, volumeID, annotations); err != nil {
+				// If updating pvc annotation fails, the volume should not be fake attached
+				return false, err
+			}
+
+			// If volume is inaccessible, it can be fake attached.
+			return volHealthStatus == common.VolHealthStatusInaccessible, nil
+		}
+		// For all other cases, return false
+		return false, nil
+	}
+	// Annotation is not found or not set to true, return false
+	log.Infof("Annotation %s not found or not set to true on pvc for volume %s", common.AnnIgnoreInaccessiblePV, volumeID)
+	return false, nil
+}
+
+// ClearFakeAttached checks if pvc corresponding to the volume has fake annotations,
+// and unmark it as not fake attached.
+func (c *K8sOrchestrator) ClearFakeAttached(ctx context.Context, volumeID string) error {
+	log := logger.GetLogger(ctx)
+	//Check pvc annotations
+	pvcAnn, err := c.getPVCAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get pvc annotations while determining if it was fake attached")
+		return err
+	}
+	val, found := pvcAnn[common.AnnFakeAttached]
+	if found && val == "yes" {
+		log.Infof("Volume: %s was fake attached", volumeID)
+		//Clear the fake attach annotation
+		annotations := make(map[string]string)
+		annotations[common.AnnFakeAttached] = ""
+		if err := c.updatePVCAnnotations(ctx, volumeID, annotations); err != nil {
+			log.Error("failed to clear fake attach annotation on the pvc")
```

Comment:
Can you print the PVC name?

Reply:
The PVC name isn't available in this method. I'm printing the PVC name in `updatePVCAnnotations` if it fails.
```
log.Errorf("failed to update pvc annotations %s/%s with err:%+v", pvcNamespace, pvcName, err)
```